### ISSUE TITLE
Search / Filtering Orders

### ISSUE THREAD

**Issue by Pauldro** [cptechinc/soft-crm#147]
Go to vendor/cptechinc/src/OrderPanel.class.php
add  a public property of filters
Make a directory called search, then a directory under that
sales-orders, and another called quotes
then make a search-orders.php and search-quotes.php under each respectively

## Make a template called order-search.php
Upload it, then in processwire make 2 page s with that template under /search one called sales orders another called quotes

```
switch ($page->name) { //$page->name is what we are editing
        case 'order':
            $ordn = $input->get->text('ordn');
            $page->body = $config->paths->content.'search/sales-orders/search-orders.php';
            break;
        case 'quote':
            $qnbr = $input->get->text('qnbr');
            $page->body = $config->paths->content.'search/quotes/search-quotes.php';
            break;
    }

include $config->paths->content."common/include-page.php";
```
Go to site/templates/content/sales-orders/search-orders.php
paste this
	```
$orderpanel = new RepSalesOrderPanel(session_id(), $page->fullURL, '#ajax-modal', '#orders-panel', $config->ajax);
	$orderpanel->pagenbr = $input->pageNum;
	$orderpanel->activeID = !empty($input->get->ordn) ? $input->get->text('ordn') : false;
	$orderpanel->get_ordercount();

	$paginator = new Paginator($orderpanel->pagenbr, $orderpanel->count, $orderpanel->pageurl->getUrl(), $orderpanel->paginationinsertafter, $orderpanel->ajaxdata);
```

Then make your filter page like you've mocked up then for the table side all you would have to do is
<?php include $config->paths->content.'salesrep/orders/orders-table.php'; ?>
<?= $paginator; ?>
When you get there let me know.

**Comment by Pauldro** [comment]
@bbullemer Pt. 2
So the next part is getting the form to generate the get URL when you submit.
You're going to add a hidden input in your form with the name filter and the value filter.
These are the names all the inputs should have

CustPO -> custpo[]
custID -> custID[] * for both from and through field
order # field -> orderno[] * for both from and through field
orderdate -> orderdate[] * for both from and through fields

Once you finish let me know and we'll test it from there.

**Comment by Pauldro** [comment]
Go to vendor/cptechinc/src/SalesOrderPanel.class.php and add this to the properties
public $filterable = array(
	'custpo' => 'between',
	'custid' => 'between',
	'orderno' => 'between',
	'orderdate' => 'between'
	'status' => 'in'
);

**Comment by Pauldro** [comment]
Fixed by 5612476901b7599b9eea2c700688f0820669a310
